packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -71,19 +76,22 @@ function concatenateAllowedTypes(fieldKey: string, fieldTypes: TreeTypeSet | und
 function getObjectAllowedTypes(schema: ObjectNodeStoredSchema): string {
 	let result = "";
 
+	// Checks if the field is the last field in the map to prevent adding a trailing comma.
+	let idx = 0;
+	const objectSize = schema.objectNodeFields.size;
+
 	for (const [fieldKey, treeFieldStoredSchema] of schema.objectNodeFields) {
 		// Set of allowed tree types `TreeTypeSet`.
 		const fieldTypes = treeFieldStoredSchema.types;
 
-		result += concatenateAllowedTypes(fieldKey, fieldTypes);
+		result += `${concatenateAllowedTypes(fieldKey, fieldTypes, idx === objectSize - 1)}`;
+		idx += 1;
```

Comment:
Instead of doing this tracking, you could update `result` to be an array of strings you push to, and at the end you could return `result,join(", ")`.

Reply:
That is so true. I should have came up with that😮‍💨